frontend/src/hooks/useRouterFilter.ts

```diff
@@ -0,0 +1,67 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import React, { useState } from 'react';
+import { saveFilter } from 'reducers/filterSlice';
+import { RootState } from 'reducers/rootReducer';
+import _ from 'lodash';
+import { BasePropertyFilter } from 'components/common/interfaces';
+
+const defaultFilter: BasePropertyFilter = {
+  searchBy: 'address',
+  pid: '',
+  address: '',
+  municipality: '',
+  projectNumber: '',
+  agencies: '',
+  classificationId: '',
+  minLotSize: '',
+  maxLotSize: '',
+};
+
+export const paramsToObject = (searchString: string) => {
+  const params = new URLSearchParams(searchString);
+  const obj: any = {};
+  for (const key of params.keys()) {
+    if (params.getAll(key).length > 1) {
+      obj[key] = params.getAll(key);
+    } else {
+      obj[key] = params.get(key);
+    }
+  }
+  return obj;
+};
+
+/**
+ * Control the state of the passed filter using url search params.
+ * NOTE: URLSearchParams not supported by IE.
+ */
+export const useRouterFilter = (filter: any, setFilter: (val: any) => void, key: string) => {
```

Comment:
Does this mean that on IE the page won't work?  Or just that they can't get to a specific filter by copying a URL?

Reply:
Haven't tested, but my guess is that if IE calls that function it will throw an exception. So year, this will likely break the map and listview for IE.

Reply:
Could add a try catch here to essentially do no filter saving for IE.

Reply:
I think we should spend a very little time to see if it doesn't barf on IE.  Although I don't know if IE can even login at this point.